From c00463d256729f1dcd2d9a5c43dde2e4eb4de129 Mon Sep 17 00:00:00 2001
From: Levi Shafter <lshafter@21sw.us>
Date: Fri, 23 Jan 2026 11:53:08 -0700
Subject: [PATCH] fix: disable bootdelay/keypress detection on Pi5-family

The Raspberry Pi 5 and Compute Module 5 seem to receive spurious
characters on the debug UART which interrupts the boot process.

Setting bootdelay to -2 prevents u-boot from stopping on the serial
console.

This is set at the top so as not to conflict with mender-specific
U-Boot RPi configuration patches

Modification of https://github.com/mendersoftware/meta-mender-community/commit/e3d4f8f555c937d370214dfd6828a65876ab3e89

Signed-off-by: Levi Shafter <lshafter@21sw.us>
---
 configs/rpi_arm64_defconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configs/rpi_arm64_defconfig b/configs/rpi_arm64_defconfig
index 9fe5d177943..9a380847e4a 100644
--- a/configs/rpi_arm64_defconfig
+++ b/configs/rpi_arm64_defconfig
@@ -1,3 +1,4 @@
+CONFIG_BOOTDELAY=-2
 CONFIG_ARM=y
 CONFIG_POSITION_INDEPENDENT=y
 CONFIG_ARCH_BCM283X=y
